deeppavlov/models/syntax_parser/joint.py

```diff
@@ -0,0 +1,142 @@
+        if not isinstance(parser_prettifier, DependencyOutputPrettifier):
+            raise ValueError("The tagger should output prettified data: last component of the config "
+                             "should be either a DependencyOutputPrettifier instance.")
+        parser_prettifier.return_string = False
+
+    def __call__(self, data: Union[List[str], List[List[str]]])\
+            -> Union[List[List[dict]], List[str], List[List[str]]]:
+        r"""Parses a batch of sentences.
+
+        Args:
+            data: either a batch of tokenized sentences, or a batch of raw sentences
+
+        Returns:
+            `answer`, a batch of parsed sentences. A sentence parse is a list of single word parses.
+            Each word parse is either a CoNLL-U-formatted string or a dictionary.
+            A sentence parse is returned either as is if ``self.to_output_string`` is ``False``,
+            or as a single string, where each word parse begins with a new string.
+
+        .. code-block:: python
+
+            >>> from deeppavlov.core.commands.infer import build_model
+            >>> model = build_model("ru_syntagrus_joint_parsing")
+            >>> batch = ["Девушка пела в церковном хоре.", "У этой задачи есть сложное решение."]
+            >>> print(*model(batch), sep="\\n\\n")
+                1	Девушка	девушка	NOUN	_	Animacy=Anim|Case=Nom|Gender=Fem|Number=Sing	2	nsubj	_	_
+                2	пела	петь	VERB	_	Aspect=Imp|Gender=Fem|Mood=Ind|Number=Sing|Tense=Past|VerbForm=Fin|Voice=Act	0	root	_	_
+                3	в	в	ADP	_	_	5	case	_	_
+                4	церковном	церковный	ADJ	_	Case=Loc|Degree=Pos|Gender=Masc|Number=Sing	5	amod	_	_
+                5	хоре	хор	NOUN	_	Animacy=Inan|Case=Loc|Gender=Masc|Number=Sing	2	obl	_	_
+                6	.	.	PUNCT	_	_	2	punct	_	_
+
+                1	У	у	ADP	_	_	3	case	_	_
+                2	этой	этот	DET	_	Case=Gen|Gender=Fem|Number=Sing	3	det	_	_
+                3	задачи	задача	NOUN	_	Animacy=Inan|Case=Gen|Gender=Fem|Number=Sing	4	obl	_	_
+                4	есть	быть	VERB	_	Aspect=Imp|Mood=Ind|Number=Sing|Person=3|Tense=Pres|VerbForm=Fin|Voice=Act	0	root	_	_
+                5	сложное	сложный	ADJ	_	Case=Nom|Degree=Pos|Gender=Neut|Number=Sing	6	amod	_	_
+                6	решение	решение	NOUN	_	Animacy=Inan|Case=Nom|Gender=Neut|Number=Sing	4	nsubj	_	_
+                7	.	.	PUNCT	_	_	4	punct	_	_
+
+            >>> # Dirty hacks to change model parameters in the code, you should do it in the configuration file.
+            >>> model["main"].to_output_string = False
+            >>> model["main"].output_format = "json"
+            >>> for sent_parse in model(batch):
+            >>>     for word_parse in sent_parse:
+            >>>         print(word_parse)
+            >>>     print("")
+                {'id': '1', 'word': 'Девушка', 'lemma': 'девушка', 'upos': 'NOUN', 'feats': 'Animacy=Anim|Case=Nom|Gender=Fem|Number=Sing', 'head': '2', 'deprel': 'nsubj'}
```

Comment:
From point of view of developer who uses results of analysis of the component it is better to make `feats` attribute as dict of key values like:
```{'Animacy': "Anim",
"Case":"Nom",
...}```
instead of string of serialized values. Because serialized string requires redundant parsing of results. Although if it is a standard format it is okay...

Reply:
It is a standard UD format.